alpha/action/diff.go

```diff
@@ -53,7 +58,11 @@ func (a Diff) Run(ctx context.Context) (*declcfg.DeclarativeConfig, error) {
 			}
 			return nil, fmt.Errorf("error rendering old refs: %v", err)
 		}
-		if err := a.MergeType.MergeDC(oldCfg); err != nil {
+
+		if err != nil {
+			return nil, fmt.Errorf("error setting merge type: %v", err)
+		}
+		if err := a.Merger.MergeDC(oldCfg); err != nil {
```

Comment:
I'm still confused by what it means to merge a single `*declcfg.DeclarativeConfig`.

Shouldn't we be expecting any given individual FBC to be valid?

 I would have expected to see something like:
```go
// Validate the old config
oldModel, err := declcfg.ToModel(oldCfg)
if err != nil {
	return nil, err
}
if err := oldModel.Validate(); err != nil {
	return nil, err
}

// Validate the new config
newModel, err := declcfg.ToModel(newCfg)
if err != nil {
	return nil, err
}
if err := newModel.Validate(); err != nil {
	return nil, err
}

// Merge them together
if err := a.Merger.Merge(oldCfg, newCfg); err != nil {
	return nil, err
}
```

Reply:
Sorry @joelanford, I think I missed the earlier comment when you mentioned this concern before. I think the use-case for keeping the single argument to the `Merge` function is that the `oldCfg` and the `newCfg` are generated by `Render`. Since `render` isn't handling duplicate objects it would be possible for the FBC created by the groups of `OldRefs` and `NewRefs` could be invalid. Would changing from `Merge` to `Normalize` make more sense here?

Reply:
> Would changing from Merge to Normalize make more sense here?

Perhaps

> `oldCfg` and the `newCfg` are generated by Render.

Makes sense, but what's the case that led to those needing to be normalized? Shouldn't old and new be valid standalone FBCs?

Reply:
What I want to avoid is having our exported libraries normalize the possibility of an FBC being invalid and us providing code that can turn an FBC from invalid to valid.

If there's something happening in `opm alpha diff` that gives rise to the need to normalize some intermediate FBC state, I think all of this is fine, but I'd probably make it internal and unexported.

Reply:
Just adding some info here for historical purposes -
- The merge functionality was public because it is required by `oc-mirror`, but we have agreed to move this functionality directly into the `oc-mirror` codebase since the requirement to merge or normalize invalid FBCs is a unique use-case.

Reply:
Can you clarify?  Is the entire merge function moving to `oc-mirror` or just the single-fbc normalization function?

Reply:
Just the single-fbc normalization work proposed on this PR was moved to `oc-mirror`